File: Capítulo_4/Dividir_para_conquistar/Exercícios/maior_valor.py
def maior_valor(lista):
    # Caso-base: lista vazia
    if len(lista) == 0:
        return None
    # Caso-base: lista com aplenas um elemento
    elif len(lista) == 1:
        return lista[0]
    else:
        return lista[0] if lista[0] > maior_valor(lista[1:]) else maior_valor(lista[1:])
    
lista = [3, 8, 1, 10, 5]

maior =  maior_valor(lista)

print(f"O maior valor na lista é: {maior}")

File: Capítulo_4/Dividir_para_conquistar/Exercícios/maior_valor.js
function maiorValor(lista) {
    if (lista.length === 0) {
        // Caso-base: lista vazia
        return null;
    } else if (lista.length === 1) {
        // Caso-base: lista com aplenas um elemento
        return lista[0];
    } else {
        return lista[0] > maiorValor(lista.slice(1))
            ? lista[0]
            : maiorValor(lista.splice(1));
    }
}

let lista = [3, 8, 1, 10, 5];
console.log(lista.length);

maior = maiorValor(lista);

console.log(`O maior valor na lista é: ${maior}`);
